### PR TITLE
Add Support for signatures on the G2 curve for BDN

### PR DESCRIPTION
**Pull request by matejpavlovic**
Feel free to further edit the code in this PR, in particular adding proper tests.

### CODE REVIEW

**Review by AnomalRoil — CHANGES_REQUESTED** [review]

File: sign/bdn/bdn.go [hunk 1]
```diff
@@ -67,29 +67,68 @@ func hashPointToR(pubs []kyber.Point) ([]kyber.Scalar, error) {
...
-func Verify(suite pairing.Suite, x kyber.Point, msg, sig []byte) error {
-	return bls.NewSchemeOnG1(suite).Verify(x, msg, sig)
+func Verify(scheme *Scheme, x kyber.Point, msg, sig []byte) error {
+	return scheme.blsScheme.Verify(x, msg, sig)
 }
 
 // AggregateSignatures aggregates the signatures using a coefficient for each
 // one of them where c = H(pk) and H: G2 -> R with R = {1, ..., 2^128}
-func AggregateSignatures(suite pairing.Suite, sigs [][]byte, mask *sign.Mask) (kyber.Point, error) {
+func AggregateSignatures(scheme *Scheme, sigs [][]byte, mask *sign.Mask) (kyber.Point, error) {
```
[the remark this answers is not in the document]
Comment (matejpavlovic) [reply sign/bdn/bdn.go]: I wanted to minimize the changes to the structure of the code as much as possible, so I just changed the name and type of the (already existing) function parameter. But it also makes sense to have those functions as methods of `Scheme`. New version ready for review.

**Comment by matejpavlovic** [comment]
I would also like to mention that my understanding of the code is still rather superficial. This shows, for example, that I did not manage to generalize the unit tests. Please make sure that the code actually works before merging. Thanks!

**Comment by Kubuxu** [comment]
Hey @AnomalRoil let me know what needs to get done to get this merged.

**Comment by AnomalRoil** [comment]
It needs more tests I guess, which has been on my to-do for like 2 months 😅

**Comment by Kubuxu** [comment]
While working on more tests, I discovered that `hashToPoint` is not implemented on bn256 G2 Point.
See the PR to upstream kyber: https://github.com/dedis/kyber/pull/428/files

This is not an issue with bls12381 we are using.

**Comment by Kubuxu** [comment]
Pull in https://github.com/Kubuxu/kyber/commit/c5d6a111d0149c40c0c845a7033c313666dd0a5c for additional tests

**Comment by AnomalRoil** [comment]
I just need to  add some retro-compat functions to avoid breaking public APIs and I'll release a new minor release with this 👍🏻

**Pull request merged**